src/main/java/com/yahoo/memory/ReadOnlyException.java

```diff
@@ -10,7 +10,7 @@
  *
  * @author Praveenkumar Venkatesan
  */
-public class ReadOnlyException extends RuntimeException {
+public class ReadOnlyException extends IllegalStateException {
```

Comment:
Illegal State could mean the application is in a bad state not due to any fault of the user.  I would suggest we extend UnsupportedOperationException -> RuntimeException.  This is what Java does with its ReadOnly exceptions.  I will fix.